andela/ah-backend-iroquois
# feat(Tags): Enable tagging of articles

**What does this PR do?** 
   - It enables users to tag their articles.

**Description of Tasks to be completed?**
Have the following endpoints working

`POST /api/articles/` 
`PUT /api/articles/<slug>/` 
`POST api/articles/tags/tag_list/`
`GET /api/articles/tags/tag_list/`
`GET api/articles/tags/tag_list/<tag_id>/`
`PUT api/articles/tags/tag_list/<tag_id>/`
`DELETe api/articles/tags/tag_list/<tag_id>/`

**How should this be manually tested?**
- After cloning the repo, cd into the directory. 
- Run ```pip install -r requirements.txt```
- Use Postman to try out the endpoints.

Only authenticated users are allowed to access the above end points. So supply the token in the header 
- To post an article with a tag(s), use the following endpoint: `POST /api/articles/`
 
```json
{
        "article": {
            "title": " The last week of sems1",
            "description": "Sims1 is ending this week and it has been intersting",
            "body": "I had enough time to make this short.",
            "tags": ["Django", "Python", "Java", "SpringBoot", "Hibernate", "PrimeFaces"]
        }
    }
```
- To update a tag on a post, use the following endpoint: `POST api/articles/tags/<slug>/`

```json
{
  "article": {
    "title": "How to train your dragon",
    "description": "Ever Wonder how?",
    "body": "Keep reading...",
    "tags": ["tag2", "tag3"]
  }
}
```

- To post a tag, use the following endpoint: `POST api/articles/tags/tag_list/`

```json
{
	"tag_name": "politics"
}
```

- To get a list of tags, use the following endpoint: `GET /api/articles/tags/tag_list/`. 
- To get, update and delete a single tag, use the following endpoint: `api/articles/tags/tag_list/<tag_id>/`. You should be a super user.

**What are the relevant pivotal tracker stories?**

- [Finishes #159952014](https://www.pivotaltracker.com/story/show/159952014)
